### PR TITLE
Fix D3 backend Error

### LINKED ISSUE #845: D3 backend  Error
ploomber report --> gives me an error , ValueError: Error when using d3 backend: expected a path with extension .html, but got: '/tmp/tmpmojk06x4.png', please change the extension

### PR DESCRIPTION
**Pull request by 94rain**
## Describe your changes
* add `--backend` argument to `report` command
* add `backend` argument to `dag.to_markup` and `dag.plot`
* add `dot -c` in Windows CIs to activate png on Windows (according to https://stackoverflow.com/questions/35688320/graphviz-seems-doesnt-support-png-and-map)

## Issue ticket number and link
Closes #845

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have added thorough tests (when necessary).
- [x] I have added the right documentation (when needed). Product update? If yes, write one line about this update.

### CODE REVIEW

File: src/ploomber/cli/report.py [hunk 1]
```diff
@@ -14,8 +16,16 @@ def main():
             '-o',
             help='Where to save the report, defaults to pipeline.html',
             default='pipeline.html')
+        parser.add_argument(
+            '--backend',
+            '-b',
+            help=('How to generate the plot: d3 or '
+                  'pygraphviz. Using whatever is '
+                  'available by default'),
```
Comment (idomic) [inline src/ploomber/cli/report.py:24]: > How to generate the plot: d3 or pygraphviz. Using whatever is available by default

Maybe Decides on the backend to generate the plot: d3 or pygraphviz, the one available by default.

Comment (94rain) [reply src/ploomber/cli/report.py:24]: Updated

File: tests/cli/test_custom.py [hunk 2]
```diff
@@ -531,3 +531,21 @@ def load(product):
...
+@pytest.mark.parametrize('output', [None, 'pipeline.html', 'pipeline.png'])
+@pytest.mark.parametrize('backend', [None, 'd3', 'pygraphviz'])
+def test_report_command(tmp_nbs, backend, output, monkeypatch, capsys):
+    args = ['ploomber', 'report', '--entry-point', 'pipeline.yaml',
+            '--backend', backend]
+    if output is not None:
+        args.extend(['--output', output])
+    monkeypatch.setattr(
+        sys, 'argv', args)
+    output = 'pipeline.html' if output is None else output
```
Comment (idomic) [inline tests/cli/test_custom.py:545]: Same as above

**Review by idomic — CHANGES_REQUESTED** [review]
Please fix, 2 small things

**Review by idomic — APPROVED** [review]
lgtm

File: tests/cli/test_custom.py [hunk 3]
```diff
@@ -531,3 +531,21 @@ def load(product):
...
                 'TaskBuildError: Error building task "load"')
     assert expected in captured.err
+
+
+@pytest.mark.parametrize('output', [None, 'pipeline.html', 'pipeline.png'])
+@pytest.mark.parametrize('backend', [None, 'd3', 'pygraphviz'])
+def test_report_command(tmp_nbs, backend, output, monkeypatch, capsys):
+    args = ['ploomber', 'report', '--entry-point', 'pipeline.yaml',
+            '--backend', backend]
+    if output is not None:
```
[the remark this answers is not in the document]
Comment (94rain) [reply tests/cli/test_custom.py]: Updated

**Comment by 94rain** [comment]
The failing test (only on Linux 3.9) might be intermittent as well. `pyppeteer.errors.TimeoutError: Navigation Timeout Exceeded: 8000 ms exceeded.`
https://github.com/ploomber/ploomber/runs/7341058539?check_suite_focus=true#step:8:316

**Comment by idomic** [comment]
Did some manual tests, nice job!

**Comment by idomic** [comment]
@94rain if you'll look in the report it's still missing the plot, please open a PR to make it show:
Screen Shot 2022-07-19 at 13 03 20

**Pull request merged**